R/SeuratUtils.R

```diff
@@ -104,7 +104,9 @@ SplitSeurat <- function(seuratObj, splitField, minCellsToKeep = 0.02, naOtherLab
 		minCellsToKeep <- ncol(seuratObj) * minCellsToKeep
 		print(paste0('Interpreting minCellsToKeep as a fraction of input cells. Converting from ', minCellsToKeepOrig, ' to: ', minCellsToKeep))
 	}
-
+  #This prevents NAs in the currently set Idents field from preventing cells.idents from properly splitting the cells.
+  Seurat::Idents(seuratObj) <- splitField 
```

Comment:
do we really need to do this? all the subsets are using cells='', so we dont care about Idents(), do we?

Reply:
I don't understand the exact underlying cause, but I think `subset(seuratObj, cells = cells)` parses the current Ident to find "cells.ident", which causes those cells to be dropped if the `cells.ident` is `NA`. the unexpected behavior certainly exists, but whether we need to do it or not is a fair question. 

I think it's a minor-but-important edge case that can be fixed by a single line, so probably should.

Reply:
although, we should probably reset the ident back to its original value. I'll figure out how to do that.

Reply:
i think you could capture a vector with Idents(seuratObj), right? what about this, since i'm picky about things like this:
- Make a new util method called .FixAndWarnNaIdents(). This tests is the seurat object actually has NAs in Idents(). If no, just go on. If yes, let's set them to something like 'MISSING'.

Reply:
I can get behind this. I'll add it.

Reply:
just to put this on record: the NA splitting issue is caused by this: https://github.com/mojaveazure/seurat-object/pull/80 and should be resolved when that PR gets accepted.